### ISSUE TITLE
Prelude OptionAsync does not unwrap Nullable value when created

### ISSUE THREAD

**Issue by Lonli-Lokli** [louthy/language-ext#816]
```
public static OptionAsync<T> OptionalAsync<T>(Task<T> value) => 
    new MOptionAsync<T>().ReturnAsync((Func<Unit, Task<T>>) (_ => value));
```
Should be probably
```
public static OptionAsync<T> OptionalAsync<T>(Task<T?> value) => 
    new MOptionAsync<T>().ReturnAsync((Func<Unit, Task<T>>) (_ => value));
```

That's important when Nullable is enabled in project.

**Comment by TysonMN** [comment]
Can you wrap both blocks of code in triple back ticks?  It will make them easier to read.

**Comment by TysonMN** [comment]
Is this issue a special case of #612?

**Comment by Lonli-Lokli** [comment]
@TysonMN It's at least related to it.
We have a Nullable-enabled project so OptionalAsync for Task<T?> still leads to T? later in chain.

**Comment by eldon111** [comment]
I just ran up against this as well.  My project isn't Nullable-enabled, but when I call `OptionalAsync(someTask)` and the task returns null, I expected it to resolve to a None, but it throws a ValueIsNullException.

**Comment by harrhp** [comment]
@eldon111 it was fixed in #799
